controllers/operatorconfigmap_controller.go

```diff
@@ -210,10 +219,10 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		},
 	}
 	err = c.createOrUpdate(c.cephFSDeployment, func() error {
+		csi.GetCephFSDeployment(c.OperatorNamespace).DeepCopyInto(c.cephFSDeployment)
```

Comment:
@bernerhat any specific reason to move this here, please put this kind of changes to its own commit so that we can understand what is the bug

@nb-ohad i think it was done to ensure that we update the deployment Template (for new changes)/update

Reply:
@Madhu-1 That does not make a lot of sense. 
`GetCephFSDeployment ` already creates a copy of the top-level structure from the template. If the concern is that somewhere in the nested structure there is a pointer to a shared sub-struct then I would suggest that `GetCephFSDeployment` should always create a deep copy before applying any logic and before returning it. 

With the current code, it seems that we are calling `deepCopy` every time `GetCephFSDeployment` is invoked which, to me, indicates that it should be part of the `GetCephFSDeployment` operation itself.

Reply:
@Madhu-1 this code was moved as a suggestion from @leelavg . we suspect the `deepCopy` is messing the ownership assignment (right below it) and causing the leftover issue for these components

Reply:
1. I couldn't understand why the DeepCopyInto exists in the mutate func in the first place, we could just update the desired spec and call it a day unless I'm missing these deployments/ds need to ensure some more which I don't know.
2. Yes I suggested to move it here and verify the bug still occurs or not.

----

I'm rephrasing what I already mentioned in https://github.com/red-hat-storage/ocs-client-operator/pull/64#pullrequestreview-1848015033

- the actual bug is two fold, one, csi deployments & ds are not getting deleted, two, CSIAddonNode finalizer is not removed & garbage collected
- for first bug, `own`ing resource after deep copy should be the fix (I asked Amit to verify this some time back)
- for second bug, we aren't the owner of CSIAddonNode, this morning I saw Madhu updated the BZ to delete csi-addons after deleting ocs-client-operator and that should be tested along w/ above (moving line)

that's it, we have a bug and an unverified fix w/ a possible root cause.

---

on redesign:

- I rightfully agree that configmap should back this controller but a bit sceptical whether to club that in this PR or not unless my suggestion above is wrong which mandates a redesign
- commented on configmap [here](https://github.com/red-hat-storage/ocs-client-operator/pull/64#discussion_r1471075608)

Reply:
@nb-ohad discussed offline with @Madhu-1 and decided to move the `deepCopy` into each ceph get function